### PR TITLE
Guide rate nnn

### PR DESCRIPTION
**Pull request by joakim-hove**
More changes to the GuideRate implementation - this code still suffers from the fact we - as in @totto82 and myself do not really understand what we are doing or where we are going. But this should at least be a step in the right direction. 

Totally safe to merge - as the code is not in use yet.

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by totto82** [comment]
During testing I noticed some issues. I have patches that solves some of them. But I wasn't able to finish the work today. I plan to make a PR to your branch, maybe on Monday.

**Comment by joakim-hove** [comment]
@totto82 : I cherry-picked your commit and rebased on upstream. Do you consider this "ready" now?

**Comment by joakim-hove** [comment]
jenkins build this please

File: src/opm/parser/eclipse/EclipseState/Schedule/Group/GuideRate.cpp [hunk 1]
```diff
@@ -22,74 +22,158 @@
 
 namespace Opm {
 
+double GuideRate::GuideRateValue::eval(GuideRateModel::Target target_arg) const
```
Comment (bska) [inline src/opm/parser/eclipse/EclipseState/Schedule/Group/GuideRate.cpp]: The `target_arg` parameter is [unused](https://ci.opm-project.org/job/opm-common-PR-builder/lastBuild/gcc/source.48458f53-bdc6-471e-8bf9-735dbe1de789/#25) in this function.  Is that intentional?

**Comment by totto82** [comment]
> @totto82 : I cherry-picked your commit and rebased on upstream. Do you consider this "ready" now?

It works as intended on the simple problems I have tested. There is at-least one thing that needs to be fixed either in the opm-common code or in the simulator. Currently it blows up if you pass it zero potentials. This can happen if the well is shut or stopped from various reasons or if the well potential calculations fails. I suggest to guard against this in the eval method in GuideRateModel. Will you do it, or shall I make a new commit you can cherry-pick?

**Comment by joakim-hove** [comment]
Ok - what should happen in the case of zero potentials? Zero guide rate? I can do that.

**Comment by totto82** [comment]
Thanks. Zero guiderate is fine. Then I will need some safeguards in the simulator code as well.

File: src/opm/parser/eclipse/EclipseState/Schedule/Group/GuideRate.cpp (hunk 1 above)
Comment (joakim-hove) [reply src/opm/parser/eclipse/EclipseState/Schedule/Group/GuideRate.cpp]: Thanks - that was a bug

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
Probably a merge race - rerun jenkins

**Comment by joakim-hove** [comment]
jenkins build this please

**Pull request merged**